src/Compilers/Core/Portable/Diagnostic/Diagnostic.cs

```diff
@@ -567,6 +567,12 @@ internal virtual bool IsNotConfigurable()
         /// </summary>
         internal bool IsUnsuppressableError()
             => DefaultSeverity == DiagnosticSeverity.Error && IsNotConfigurable();
+
+        /// <summary>
+        /// Returns true if this is a unsuppressed diagnostic with an effective error severity.
+        /// </summary>
+        internal bool IsReportedError()
```

Comment:
>IsReportedError [](start = 22, length = 15)

The name feels confusing. What does it mean to be reported? A Diagnostic itself cannot know whether it was reported or not. Another way to look at this is that every Diagnostic is reported, otherwise the instance wouldn't exist. Consider renaming to something like "IsNotSuppressedError" or something similar.

Reply:
Or "IsUnsuppressedError"

---
In reply to: [313542576](https://github.com/dotnet/roslyn/pull/37948#discussion_r313542576) [](ancestors = 313542576)